elevenlabs_mcp/utils.py

```
@@ -0,0 +1,168 @@
+    id_safe = re.sub(r'[^a-zA-Z0-9_\-]', '_', id_raw)
+    if not id_safe:
+        id_safe = "unnamed"
+    output_file_name = "{0}_{1}_{2}.{3}".format(
+        tool, id_safe, datetime.now().strftime('%Y%m%d_%H%M%S'), extension)
+    output_file = (output_path / output_file_name).resolve()
+    # Ensure the file stays within the output directory
+    try:
+        output_file.relative_to(output_path.resolve())
+    except ValueError:
+        raise ElevenLabsMcpError(
+            "Generated filename escapes output directory")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)